app/ui-message/client/messageBox/messageBox.js

```diff
@@ -400,11 +401,14 @@ Template.messageBox.events({
 
 		const files = items
 			.filter((item) => item.kind === 'file' && item.type.indexOf('image/') !== -1)
-			.map((item) => ({
-				file: item.getAsFile(),
-				name: `Clipboard - ${ moment().format(settings.get('Message_TimeAndDateFormat')) }`,
-			}))
-			.filter(({ file }) => file !== null);
+			.map((item) => {
+				const fileItem = item.getAsFile();
+
+				return {
+					file: fileItem,
+					name: `Clipboard - ${ moment().format(settings.get('Message_TimeAndDateFormat')) }.${ getImageExtensionFromMime(fileItem.type) }`,
```

Comment:
It's missing the case where `getImageExtensionFromMime(fileItem.type)` might be `undefined`.